src/lib/agent-runner.ts

```diff
@@ -0,0 +1,368 @@
+import type { FrameworkConfig } from './framework-config';
+import type { WizardOptions } from '../utils/types';
+import {
+  abort,
+  askForAIConsent,
+  confirmContinueIfNoOrDirtyGitRepo,
+  ensurePackageIsInstalled,
+  getOrAskForProjectData,
+  getPackageDotJson,
+  isUsingTypeScript,
+  printWelcome,
+  askForCloudRegion,
+} from '../utils/clack-utils';
+import { analytics } from '../utils/analytics';
+import { WIZARD_INTERACTION_EVENT_NAME } from './constants';
+import clack from '../utils/clack';
+import {
+  initializeAgent,
+  runAgent,
+  AgentSignals,
+  AgentErrorType,
+} from './agent-interface';
+import { getCloudUrlFromRegion } from '../utils/urls';
+import chalk from 'chalk';
+import {
+  addMCPServerToClientsStep,
+  uploadEnvironmentVariablesStep,
+} from '../steps';
+import * as fs from 'fs';
+import path from 'path';
+
+/**
+ * Universal agent-powered wizard runner.
+ * Handles the complete flow for any framework using PostHog MCP integration.
+ */
+export async function runAgentWizard(
+  config: FrameworkConfig,
+  options: WizardOptions,
+): Promise<void> {
+  // Setup phase
+  printWelcome({ wizardName: config.ui.welcomeMessage });
+
+  clack.log.info(
+    `🧙 The wizard has chosen you to try the next-generation agent integration for ${config.metadata.name}.\n\nStand by for the good stuff, and let me know how it goes:\n\ndanilo@posthog.com`,
+  );
+
+  const aiConsent = await askForAIConsent(options);
+  if (!aiConsent) {
+    await abort(config.metadata.abortMessage, 0);
+  }
+
+  const cloudRegion = options.cloudRegion ?? (await askForCloudRegion());
+  const typeScriptDetected = isUsingTypeScript(options);
+
+  await confirmContinueIfNoOrDirtyGitRepo(options);
+
+  // Framework detection and version
+  const packageJson = await getPackageDotJson(options);
+  await ensurePackageIsInstalled(
+    packageJson,
+    config.detection.packageName,
+    config.detection.packageDisplayName,
+  );
+
+  const frameworkVersion = config.detection.getVersion(packageJson);
```

Comment:
I think this is the part we gotta build out for handling things like react. Some way to configure a whole set of decisions to take/make to arrive at the exact combo of react + router option + store(redux, no redux, etc.).